### PR TITLE
Create a separate requirements file for the CI

### PR DESCRIPTION
**Pull request by sduquemesa**
**Context:** The requirements.txt file has packages unpinned. This is contrary to the convention of pinning packages in the requirements.txt file.

**Description of the Change:** Create a separate `requirements.txt` (pinned) and `requirements-ci.txt` (unpinned). This latter is used by the CI.

**Benefits:** CI environment will behave closely as when installing the package with `setup.py`

**Possible Drawbacks:** Version conflicts on (unpinned) dependencies potentially will only appear on CI environment.

**Related GitHub Issues:** #582

### CODE REVIEW

File: requirements-ci.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+numpy
+scipy
+sympy
+tensorflow
+tensorboard
+networkx
+git+https://github.com/XanaduAI/blackbird@master#egg=quantum-blackbird
```
Comment (thisac) [inline requirements-ci.txt:7]: We should be able to use the released version of Blackbird now.
```suggestion
quantum-blackbird
```

File: requirements.txt [hunk 2]
```diff
@@ -1,14 +1,14 @@
 numpy>=1.19.2
-scipy
+scipy>=1.7.1
 sympy>=1.5
 tensorflow>=2.0
 tensorboard>=2.0
 networkx>=2.0
 git+https://github.com/XanaduAI/blackbird@master#egg=quantum-blackbird
```
Comment (thisac) [inline requirements.txt:7]: Same here!
```suggestion
quantum-blackbird==0.4.0
```

**Review by thisac — CHANGES_REQUESTED** [review]
Thanks @sduquemesa. I'd think we should strictly pin all dependencies in the `requirements.txt` file to latest releases (e.g. `numpy==1.21` instead of `numpy>=1.19.2`), unless already pinned (then keeping them is probably good).

I'm not sure about best practice when it comes to completely unpinning `requirements-ci.txt` (i.e. not even having a minimum version). This seems to be the case in PL, though.

**Comment by sduquemesa** [comment]
@thisac, following the discussion on issues #582, this is the reply from @antalszava:

> Yes! The idea would be to only pin only those packages to a certain version in the requirements-ci.txt file where there is a specific constraint (i.e., SF only works with a certain version of the dependency). The rest of the packages would stay unpinned so that when issuing pip install -r requirements-ci.txt, the latest version for each package is installed by the continuous integration check.

In contrast, in the requirements.txt file we would like to provide the user with a working setup to use Strawberry Fields. We provide a list of dependencies with pinned versions, such that by doing pip install -r requirements.txt, they will get a working combination of dependencies and be able to use Strawberry Fields.

**Comment by thisac** [comment]
Thanks @sduquemesa! We should update the recently merged `.github/workflows/upload.yml` file to use `pip install -r requirements-ci.txt` as well, so that we're testing with the same package versions for release. Looks good otherwise! 💯

File: requirements.txt [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 quantum-blackbird==0.4.0
 python-dateutil==2.8.0
-thewalrus>=0.16.0
-toml
-appdirs
-numba>=0.48.0
+thewalrus==0.16.2
+toml==0.10.2
+appdirs==1.4.4
+numba
```
Comment (thisac) [inline requirements.txt:13]: How come `numba` isn't pinned anymore?

File: .github/workflows/upload.yml [hunk 4]
```diff
@@ -22,7 +22,7 @@ jobs:
 
       - name: Install test dependencies
         run: |
-          pip install -r requirements.txt
+          pip install -r requirements-ci.txt
```
Comment (thisac) [inline .github/workflows/upload.yml:25]: 🎉

**Review by thisac — APPROVED** [review]
Thanks for pinging me on this @sduquemesa!

File: requirements.txt (hunk 3 above)
Comment (sduquemesa) [reply requirements.txt:13]: Sorry, my bad 😖 Fixed now!

**Pull request merged**